agent/api/task.go

```diff
@@ -190,6 +190,13 @@ func (task *Task) DockerConfig(container *Container) (*docker.Config, *DockerCli
 }
 
 func (task *Task) dockerConfig(container *Container) (*docker.Config, *DockerClientConfigError) {
+	// Detect the name for S3 images
+	dockerImage := container.Image
+	if strings.HasPrefix(dockerImage, "s3://") {
+		slice := strings.Split(dockerImage, "/")
+		dockerImage = slice[len(slice)-1]
```

Comment:
So for a given image s3://my-bucket/my-image.tar the actual image name must be my-bucket/my-image.tar?